integration-tests/tests/client/sharding_upgrade.rs

```diff
@@ -0,0 +1,274 @@
+use std::collections::HashSet;
+
+use crate::process_blocks::{create_nightshade_runtimes, set_block_protocol_version};
+use near_chain::{ChainGenesis, Provenance};
+use near_chain_configs::Genesis;
+use near_client::test_utils::TestEnv;
+use near_crypto::{InMemorySigner, KeyType, Signer};
+use near_logger_utils::init_test_logger;
+use near_primitives::account::id::AccountId;
+use near_primitives::block::Block;
+use near_primitives::epoch_manager::ShardConfig;
+use near_primitives::shard_layout::{account_id_to_shard_uid, ShardLayout};
+use near_primitives::transaction::SignedTransaction;
+use near_primitives::types::ProtocolVersion;
+use near_primitives::version::ProtocolFeature;
+use near_primitives::views::QueryRequest;
+use near_store::test_utils::gen_accounts;
+use nearcore::config::GenesisExt;
+use nearcore::NEAR_BASE;
+
+const SIMPLE_NIGHTSHADE_PROTOCOL_VERSION: ProtocolVersion =
+    ProtocolFeature::SimpleNightshade.protocol_version();
+
+// Checks that account exists in the state after `block` is processed
+// This function checks both state_root from chunk extra and state root from chunk header, if
+// the corresponding chunk is included in the block
+fn check_account(env: &mut TestEnv, account_id: &AccountId, block: &Block) {
```

Comment:
Probably better to default to `///` for item-level comments (they become semantic and processed by `cargo doc`), but no strong feeling here.